docs/BillOfMaterials.md

```diff
@@ -0,0 +1,170 @@
+  - noship - these files should not be published to partner teams or released publically. This may include test artifacts.
+
+## A more complete example
+
+```xml
+<Build Id="(commithash)+Release+Signed+123" Created="12/1/2017 5:10:06 PM +00:00">
+
+    <Artifacts Category="ship">
+        <Artifact Id="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" Type="NuGetPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.Extensions.DependencyInjection"
+            Version="2.0.0-rtm-123"
+            BranchName="release/2.0.0"
+            CommitHash="xyz123"
+            RepositoryUrl="https://github.com/aspnet/DependencyInjection"
+            FileHash="ABCDEF0123456789" />
+        <Artifact Id="Microsoft.AspNetCore.2.0.0-rtm-123.nupkg" Type="NuGetPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.AspNetCore"
+            Version="2.0.0-rtm-123" />
+        <Artifact Id="Microsoft.AspNetCore.2.0.0-rtm-123.symbols.nupkg" Type="NuGetSymbolsPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.AspNetCore"
+            Version="2.0.0-rtm-123" />
+        <Artifact Id="aspnetcore-store-2.0.0-rtm-123-win-x64.zip" Type="ZipArchive"
+            ShouldBeSigned="false"
+            RuntimeIdentifier="win-x64"
+            Version="2.0.0-rtm-123" />
+        <Artifact Id="aspnetcore-store-2.0.0-rtm-123-linux-x64.deb" Type="DebianPackage"
+            ShouldBeSigned="false"
+            RuntimeIdentifier="linux-x64"
+            Version="2.0.0-rtm-123" />
+    </Artifacts>
+
+    <Artifacts Category="noship">
+        <Artifact Id="Experimental.1.0.0-alpha.123.nupkg" Type="NuGetPackage"
+            PackageId="Experimental"
+            Version="1.0.0-alpha.123+commithash:123456" />
+        <Artifact Id="Microsoft.AspNetCore.Tests.zip" Type="TestBundle"
+            RuntimeFramework="netcoreapp2.0"
+            RuntimeFrameworkVersion="2.0.3" />
+    </Artifacts>
+
+    <Dependencies>
+        <Link Source="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" Target="Microsoft.Extensions.Common.2.0.0.nupkg" />
+        <Link Source="Microsoft.AspNetCore.2.0.0-rtm-123.nupkg" Target="Newtonsoft.Json.9.0.1.nupkg" />
+        <Link Source="Microsoft.AspNetCore.2.0.0-rtm-123.nupkg" Target="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" />
+        <Link Source="aspnetcore-store-2.0.0-rtm-123-linux-x64.deb" Target="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" />
+        <Link Source="aspnetcore-store-2.0.0-rtm-123-linux-x64.deb" Target="Microsoft.AspNetCore.2.0.0-rtm-123.nupkg" />
+    </Dependencies>
+
+</Build>
+```
+
+### Example usage: signing
+
+In the example above, some of the `<Artifact>` items were marked `ShouldBeSigned="true"`. Our signing tool could use this as a way to
+determine which files should be passed on to signing.
+
+### Example usage: metadata
```

Comment:
Do we need mechanisms for appending to a bom or merging multiple boms for ingestion? For example, the runtime store installers run after UniverseCoherence. How do we coalesce the artifacts from those two builds for example?

Reply:
Yes, appending, merging, etc are all good tools to have. As a starting point, we could throw if two boms being merged have clashing artifact IDs.